src/utils/types.ts

```diff
@@ -826,8 +843,14 @@ export type FilterTypeOptions = {
   namedDates?: NamedDates
   // For boolean to show on and of criteria
   booleanMapping?: BooleanFilterMapping
-  // For Data View filter lists
-  getFilterList?: (table: string, column: string) => Promise<string[]>
+  // For Data View filters
+  column?: string
+  code?: string
+  dataType?: string
+  showFilterList?: boolean
+  searchFields?: string[]
+  delimiter?: string
+  nullString?: string
```

Comment:
This is horrible -- and really should be updated to union types for better type checking within each filter type.

Reply:
Actually, I didn't really change anything here -- I just made the input props for each filter stricter (removed the `any`s)